### DISCUSSION TITLE
Import error running to_coco.py

### DISCUSSION THREAD

**Discussion by punyuenyi** [bdd100k/bdd100k#189] [Q&A]
I got this error when trying to run
`!python -m bdd100k.label.to_coco -i /bdd100k/label/bdd100k_labels_images_train.json -o /bdd100k/label/det_v2_train_release_coco.json --remove-ignore`

`Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/content/bdd100k/bdd100k/label/to_coco.py", line 12, in <module>
    from scalabel.common.parallel import NPROC
ImportError: cannot import name 'NPROC' from 'scalabel.common.parallel' (/usr/local/lib/python3.7/dist-packages/scalabel/common/parallel.py)`

I have previously `pip install scalabel`
Any help will be appreciated. Ta

**Comment by jonbr407** [comment]
I had the same error the other day when trying to convert. I solved it by:

1. installing the requirements.txt after cloning the repo
2. installing the setup.py next
3. installing and updating the following: install -U nanoid numpy pydantic boto3

I hope this helps.

**Reply by punyuenyi** [reply]
> I had the same error the other day when trying to convert. I solved it by:
> 
> 1. installing the requirements.txt after cloning the repo
> 2. installing the setup.py next
> 3. installing and updating the following: install -U nanoid numpy pydantic boto3
> 
> I hope this helps.

Thank you for your reply!
I have tried your solution and it does solve the import error.
However, assertion error shows up right after the program started converting the format.
Did you have any experience with this error? Thank you!


`[2021-11-08 13:28:24,182 to_coco.py:537 main] Loading annotations...`
`[2021-11-08 13:28:35,198 to_coco.py:546 main] Start format converting...`
`  0%|                                                                     | 0/10000 [00:00<?, ?it/s]`
`Traceback (most recent call last):`
`  File "/opt/anaconda3/lib/python3.8/runpy.py", line 194, in _run_module_as_main`
`    return _run_code(code, main_globals, None,`
`  File "/opt/anaconda3/lib/python3.8/runpy.py", line 87, in _run_code`
`    exec(code, run_globals)`
`  File "/Users/yuenyipun/Desktop/bdd100k/bdd100k/label/to_coco.py", line 557, in <module>`
`    main()`
`  File "/Users/yuenyipun/Desktop/bdd100k/bdd100k/label/to_coco.py", line 547, in main`
`    frames = bdd100k_to_scalabel(dataset.frames, bdd100k_config)`
`  File "/Users/yuenyipun/Desktop/bdd100k/bdd100k/label/to_scalabel.py", line 57, in bdd100k_to_scalabel`
`    label = deal_bdd100k_category(`
`  File "/Users/yuenyipun/Desktop/bdd100k/bdd100k/label/to_scalabel.py", line 35, in deal_bdd100k_category`
`   assert category_name in bdd100k_config.ignored_mapping`
`AssertionError`

**Reply by jonbr407** [reply]
Sorry, I'm not familiar with this error, however it looks to be quite specific so there should be a lot more help documentation online for it. I'm sorry I could not have been more helpful. Goodluck! :)

**Reply by Nihal987** [reply]
Hi @punyuenyi 
I have the same error, were you able to resolve this?
